torch/csrc/jit/backends/backend.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <ATen/core/stack.h>
+
+namespace torch {
+namespace jit {
+
+// This struct represents an external backend that can be used to
+// execute JIT subgraphs.
+struct TORCH_API Backend {
+  Backend() = default;
+  /// The name of the backend.
+  std::string name;
+  /// A function that should be invoked to convert the JIT graph into
+  /// something that the backend can compile.
+  std::function<int(Stack&)> preprocess;
+  /// A function that should be invoked to compile the JIT graph.
+  std::function<int(Stack&)> compile;
+  /// A function that should be invoked to execute a previously compiled
+  /// JIT subgraph on the backend.
+  std::function<int(Stack&)> execute;
+};
+
+// Static registration API for backends.
+struct TORCH_API RegisterBackend {
```

Comment:
If these were bound in as classes then there would be no need to have a separate global registry of backends. Instead each backend could just register its class with the already existing class registry.